pkg/api/device_req_resp.go

```diff
@@ -0,0 +1,32 @@
+package api
+
+import "github.com/pborman/uuid"
+
+const (
+	// DeviceEnabled represents enabled
+	DeviceEnabled = "Enabled"
+
+	// DeviceDisabled represents disabled
+	DeviceDisabled = "Disabled"
+
+	// DeviceFailed represents failed
+	DeviceFailed = "Failed"
+)
+
+// AddDeviceReq structure
+type AddDeviceReq struct {
+	PeerID uuid.UUID `json:"peer-id"`
+	Names  []string  `json:"names"`
+}
+
+// Device is the added device info
+type Device struct {
+	PeerID uuid.UUID `json:"peer-id"`
+	Detail []Info    `json:"device-details"`
```

Comment:
You can put Name and State directly in the `Device` struct. And there shall be one `Device` struct per device.

Reply:
@prashanthpai Thats what I did earlier, which I guess I changed after our discussion.
Can you explain how should I manage the peerid detail for that device?